### PR TITLE
buildkite-test-collector-rust: 0.1.0 -> 0.1.3

### LINKED ISSUE #34: Current release `0.1.2` does not have patched lockfile
## Description

The current release (`0.1.2`) does not contain the fix from #7. This prevents users from using a frozen lockfile to build the collector.

## Reproduction

You can reproduce the issue by executing the following:

```shell
git checkout 0.1.2
cargo build --frozen --release
```

### PR DESCRIPTION
**Pull request by nickgerace**
## Description of changes

This PR updates [buildkite-test-collector](https://github.com/buildkite/test-collector-rust) to its latest version. It also adds the auto update script for future releases.

~~However, this PR is blocked until there's a new release that fixes buildkite/test-collector-rust#34. Yes, we can get around it by patching in our own lockfile, but waiting for a clean release will help with auto updates in the future.~~

EDIT: the blocking issue has been resolved.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/continuous-integration/buildkite-test-collector-rust/default.nix [hunk 1]
```diff
@@ -3,24 +3,27 @@
...
   };
 
   buildInputs = lib.optionals stdenv.isDarwin [
     Security
   ];
 
-  cargoSha256 = "sha256-qfJ0ROi0S0mmPl6kKrW3dp3VLjYqK+sBVj+iKDNTjyM=";
+  cargoSha256 = "sha256-OeuP/xkmwPOSSjd9hHvtYpOutjiXEn2BWoDQMgqmf6I=";
+
+  passthru.updateScript = nix-update-script { };
```
Comment (nickgerace) [inline pkgs/development/tools/continuous-integration/buildkite-test-collector-rust/default.nix:26]: Auto-updates for the future

File: pkgs/development/tools/continuous-integration/buildkite-test-collector-rust/default.nix [hunk 2]
```diff
@@ -3,24 +3,27 @@
...
   pname = "buildkite-test-collector-rust";
-  version = "0.1.0";
+  version = "0.1.2";
 
   src = fetchFromGitHub {
     owner = "buildkite";
     repo = "test-collector-rust";
-    rev = "v${version}";
-    sha256 = "sha256-rY/+AwxO0+xcnRj0A8TRhCUJQ0ecosybI6It1mDOdQM=";
+    rev = version;
```
Comment (nickgerace) [inline pkgs/development/tools/continuous-integration/buildkite-test-collector-rust/default.nix:16]: No longer has `v` in the version string

**Comment by nickgerace** [comment]
Keeping in drafts until https://github.com/buildkite/test-collector-rust/issues/34 is resolved.

**Comment by nickgerace** [comment]
Moving back to drafts because this does not build yet and I need to change the commit message title to match the package name.

**Comment by nickgerace** [comment]
Okay! I think we are good.

**Comment by cole-h** [comment]
(If you rebase on the latest `master`, the eval issue should be fixed)

**Comment by nickgerace** [comment]
Closing in favor of #299828

**Pull request closed without merging**